src/analyzer/analyzerbeats.cpp

```diff
@@ -138,89 +138,75 @@ bool AnalyzerBeats::isDisabledOrLoadStoredSuccess(TrackPointer tio) const {
             // sane, re-analyzing will do nothing.
-            return true;
-        } else if (m_bPreferencesReanalyzeOldBpm) {
             return false;
-        } else if (pBeats->getBpm() == 0.0) {
-            qDebug() << "BPM is 0 for track so re-analyzing despite preference settings.";
+        }
+        if (!m_bPreferencesReanalyzeOldBpm) {
             return false;
+        }
+        if (pBeats->getBpm() == 0.0) {
+            qDebug() << "BPM is 0 for track so re-analyzing despite preference settings.";
         } else if (pBeats->findNextBeat(0) <= 0.0) {
             qDebug() << "First beat is 0 for grid so analyzing track to find first beat.";
-            return false;
         } else {
             qDebug() << "Beat calculation skips analyzing because the track has"
                      << "a BPM computed by a previous Mixxx version and user"
                      << "preferences indicate we should not change it.";
-            return true;
+            return false;
         }
-    } else {
-        // If we got here, we want to analyze this track.
-        return false;
     }
+    return true;
 }
 
-void AnalyzerBeats::process(const CSAMPLE *pIn, const int iLen) {
-    if (!m_pPlugin) {
-        return;
+bool AnalyzerBeats::process(const CSAMPLE *pIn, const int iLen) {
+    VERIFY_OR_DEBUG_ASSERT(m_pPlugin) {
+        return false;
     }
 
     m_iCurrentSample += iLen;
     if (m_iCurrentSample > m_iMaxSamplesToProcess) {
-        return;
+        return true; // silently ignore all remaining samples
     }
 
-    bool success = m_pPlugin->process(pIn, iLen);
-    if (!success) {
-        m_pPlugin.reset();
-    }
+    return m_pPlugin->process(pIn, iLen);
 }
 
-void AnalyzerBeats::cleanup(TrackPointer tio) {
-    Q_UNUSED(tio);
+void AnalyzerBeats::cleanup() {
     m_pPlugin.reset();
 }
 
 void AnalyzerBeats::finalize(TrackPointer tio) {
-    if (!m_pPlugin) {
+    VERIFY_OR_DEBUG_ASSERT(m_pPlugin) {
         return;
     }
 
-    bool success = m_pPlugin->finalize();
-    qDebug() << "Beat Calculation" << (success ? "complete" : "failed");
-
-    if (!success) {
-        m_pPlugin.reset();
+    if (!m_pPlugin->finalize()) {
+        qWarning() << "Beat/BPM analysis failed";
         return;
     }
 
     BeatsPointer pBeats;
     if (m_pPlugin->supportsBeatTracking()) {
         QVector<double> beats = m_pPlugin->getBeats();
         QHash<QString, QString> extraVersionInfo = getExtraVersionInfo(
-            m_pluginId, m_bPreferencesFastAnalysis);
+                m_pluginId, m_bPreferencesFastAnalysis);
         pBeats = BeatFactory::makePreferredBeats(
-            *tio, beats, extraVersionInfo,
-            m_bPreferencesFixedTempo, m_bPreferencesOffsetCorrection,
-            m_iSampleRate, m_iTotalSamples,
-            m_iMinBpm, m_iMaxBpm);
+                *tio, beats, extraVersionInfo, m_bPreferencesFixedTempo, m_bPreferencesOffsetCorrection, m_iSampleRate, m_iTotalSamples, m_iMinBpm, m_iMaxBpm);
```

Comment:
This line is now way too long.

Reply:
After adding a single line break between the arguments the auto-formatter splits them all on separate lines. That's reasonable.